src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
+    private static Object[] stack;
+    private static int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEAFULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) throws RuntimeException {
+        if (size <= stack.length) {
+            stack[size] = value;
+            size++;
+        } else {
+            throw new StackOverflowError();
+        }
     }
 
     public T peek() {
-        return null;
+        if (size > 0) {
+            return (T) stack[size - 1];
+        } else {
+            throw new EmptyStackException();
+        }
     }
 
     public T pop() {
-        return null;
+        if (size > 0) {
+            Object value = stack[size - 1];
+            stack[size] = null;
```

Comment:
There is an off-by-one error here. You should set 'stack[size - 1]' to null instead of 'stack[size]'.